File: apollo-compiler/src/test/graphql/com/example/custom_scalar_type_warnings/custom_scalar_type_warnings/type/CustomType.java
package com.example.custom_scalar_type_warnings.type;

import com.apollographql.apollo.api.ScalarType;
import java.lang.Class;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.lang.SuppressWarnings;
import javax.annotation.Generated;

@Generated("Apollo GraphQL")
public enum CustomType implements ScalarType {
  URL {
    @Override
    public String typeName() {
      return "URL";
    }

    @SuppressWarnings("rawtypes")
    @Override
    public Class javaType() {
      return Object.class;
    }
  },

  ID {
    @Override
    public String typeName() {
      return "ID";
    }

    @SuppressWarnings("rawtypes")
    @Override
    public Class javaType() {
      return String.class;
    }
  }
}

File: apollo-compiler/src/test/graphql/com/example/custom_scalar_type_warnings/type/CustomType.kt
package com.example.custom_scalar_type_warnings.type

import com.apollographql.apollo.api.ScalarType
import java.lang.Class
import javax.annotation.Generated
import kotlin.String

@Generated("Apollo GraphQL")
enum class CustomType : ScalarType {
    URL {
        override fun typeName(): String = "URL"

        override fun javaType(): Class<*> = java.lang.Object::class.java
    },

    ID {
        override fun typeName(): String = "ID"

        override fun javaType(): Class<*> = String::class.java
    }
}
